src/components/Calculator.js

```diff
@@ -1,13 +1,67 @@
-import React from 'react';
-import Content from './Content';
+import React, { useState } from 'react';
+import calculate from '../logic/calculator';
 
 const Calculator = () => (
-  <div className="container">
-    <form>
-      <input type="text" value="" placeholder="0" />
-    </form>
-    <Content />
-  </div>
+  <Content />
 );
+const Content = () => {
+  const [state, setState] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
+
+  const handleClick = (e) => {
+    const show = () => calculate(state, e.target.name);
+    setState(show);
+  };
+
+  let result;
+  const { total, next, operation } = state;
+  if (total === null && next === null && operation === null) {
+    result = 0;
+  }
+  if (next !== null && total === null) {
+    result = next;
+  }
+
+  if (next === null && total !== null && operation !== null) {
+    result = total + operation;
+  }
+  if (next !== null && total !== null && operation !== null) {
+    result = total + operation + next;
+  }
+  if (next === null && total !== null && operation === null) {
+    result = total;
+  }
+  return (
+    <div className="container">
+      <form>
+        <input type="text" value={result} placeholder="0" />
+      </form>
+      <div className="keypad">
+        <button type="button" onClick={handleClick} name="AC">AC</button>
+        <button type="button" onClick={handleClick} name="+/-">+/-</button>
+        <button type="button" onClick={handleClick} name="%">%</button>
+        <button type="button" className="highlight" onClick={handleClick} name="/">&divide;</button>
+        <button type="button" onClick={handleClick} name="7">7</button>
+        <button type="button" onClick={handleClick} name="8">8</button>
+        <button type="button" onClick={handleClick} name="9">9</button>
+        <button type="button" className="highlight" onClick={handleClick} name="*">&times;</button>
```

Comment:
- [ ] I noticed that during the operation of multiplication (X) and division (/), clicking on the equal sign (=) to display the result of the operation, makes all the content of the page disappear. Kindly correcting that 👍🏻